packages/flutter_tools/test/build_runner/multiroot_asset_reader_test.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:build/build.dart';
+import 'package:build_runner_core/build_runner_core.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/build_runner/web_compilation_delegate.dart';
+
+import '../src/common.dart';
+import '../src/testbed.dart';
+
+void main() {
```

Comment:
Also test filtering out of `._`?

Reply:
We can't test this yet because it is only via the path that invokes real file system code.

To handle this, I'm writing a new implementation of Glob, AssetReader, and AssetWriter but that will take some time.